addons/hr_holidays/static/src/views/calendar/common/calendar_common_popover.js

```diff
@@ -1,25 +1,38 @@
-/** @odoo-module */
-
 import { CalendarCommonPopover } from "@web/views/calendar/calendar_common/calendar_common_popover";
-
+import { onWillStart } from "@odoo/owl";
+import { user } from "@web/core/user";
 import { useService } from "@web/core/utils/hooks";
 
 export class TimeOffCalendarCommonPopover extends CalendarCommonPopover {
+    static subTemplates = {
+        ...CalendarCommonPopover.subTemplates,
+        footer: "hr_holidays.TimeOffCalendarCommonPopover.footer",
+    };
+
     setup() {
         super.setup();
-
-        this.dialog = useService('dialog');
-        this.action = useService('action');
+        this.orm = useService("orm");
+        this.viewType = "calendar";
+        onWillStart(async () => {
+            this.record = this.props.record.rawRecord
+            this.state = this.record.state;
+            this.isManager = await user.hasGroup("hr_holidays.group_hr_holidays_user") || this.record.leave_manager_id?.[0] == user.userId;
+        });
```

Comment:
Add parentheses around `await user.hasGroup()` to be sure it receives the value before doing the `||`.
Also, use "===" when checking equality in javascript
here I think it does not change anything but in certain case it could do wrong things.

```suggestion
            this.isManager = (await user.hasGroup("hr_holidays.group_hr_holidays_user")) || this.record.leave_manager_id?.[0] === user.userId;
        });
```